ci/test/mkpipeline.py

```diff
@@ -53,6 +53,13 @@ def main() -> int:
     os.environ["GIT_PAGER"] = ""
     spawn.runv(["git", "diff", "--stat", "origin/main..."])
 
+    # Merge base branch in for pull request.
+    if os.environ["BUILDKITE_PULL_REQUEST"]:
+        spawn.runv(["git", "config", "--global", "user.email", "buildkite@buildkite.com"])
+        spawn.runv(["git", "config", "--global", "user.name", "Buildkite"])
+        base_branch = os.environ["BUILDKITE_PULL_REQUEST_BASE_BRANCH"]
+        spawn.runv(["git", "merge", f"origin/{base_branch}"])
```

Comment:
This means CI won't run anymore when there are conflicts with main. Doesn't this break the workflow where you push to your draft PR until CI passes and rebase afterwards?

Also, since CI is triggered by pushes, not by merges into main, this will not rule out all merge skew. It can still happen that an incompatible PR merges in the time between the CI run and the merge of your PR.

Reply:
True on both counts. I'm not sure what approach we want to have, there are lots of possibilities here. I guess since we have merge skew rarely our current approach might be ok.